src/jwks.cc

```diff
@@ -85,43 +80,41 @@ class KeyGetter : public WithStatus {
       updateStatus(Status::JwksEcCreateKeyFail);
       return nullptr;
     }
-    BIGNUM *bn_x = createBigNumFromBase64UrlString(x);
-    BIGNUM *bn_y = createBigNumFromBase64UrlString(y);
+    bssl::UniquePtr<BIGNUM> bn_x = createBigNumFromBase64UrlString(x);
+    bssl::UniquePtr<BIGNUM> bn_y = createBigNumFromBase64UrlString(y);
     if (!bn_x || !bn_y) {
       // EC public key field x or y Base64 decode fail
       updateStatus(Status::JwksEcXorYBadBase64);
       return nullptr;
     }
 
-    if (EC_KEY_set_public_key_affine_coordinates(ec_key.get(), bn_x,
-                                                 bn_y) == 0) {
+    if (EC_KEY_set_public_key_affine_coordinates(ec_key.get(), bn_x.get(),
+                                                 bn_y.get()) == 0) {
       updateStatus(Status::JwksEcParseError);
       return nullptr;
     }
     return ec_key;
   }
 
   bssl::UniquePtr<RSA> createRsaFromJwk(const std::string& n,
-                                          const std::string& e) {
+                                        const std::string& e) {
     bssl::UniquePtr<RSA> rsa(RSA_new());
-    BIGNUM *bn_n = createBigNumFromBase64UrlString(n);
-    BIGNUM *bn_e = createBigNumFromBase64UrlString(e);
+    bssl::UniquePtr<BIGNUM> bn_n = createBigNumFromBase64UrlString(n);
+    bssl::UniquePtr<BIGNUM> bn_e = createBigNumFromBase64UrlString(e);
 
```

Comment:
If either `bn_n` or `bn_e` ends up being equal to `nullptr` at this point, I think the docs to `RSA_set0_key` say we mustn't call it [1].  I think this may only happen when `absl::WebSafeBase64Unescape` failed in `createBigNumFromBase64UrlString()`. If that can't happen, we should propose a doc comment to make it explicit that createRsaFromJwk trusts its inputs won't fail unescaping. If it can happen, we need to handle this case?

[1] https://www.openssl.org/docs/man1.1.1/man3/RSA_set0_key.html says

```
The n, e and d parameter values can be set by calling RSA_set0_key() and passing the new values for n, e and d as parameters to the function. The values n and e must be non-NULL the first time this function is called on a given RSA object. 
```

Reply:
There is already a check for either `bn_n` or `bn_e` being `nullptr`, which will prevent `RSA_set0_key()` being called in that case.

Reply:
Ah ok lgtm